src/binary_search_tree/binary_search_tree.js

```diff
@@ -0,0 +1,125 @@
+class Node {
+  constructor(data) {
+    this.data = data
+    this.left = null
+    this.right = null
+  }
+}
+
+class BinarySearchTree {
+  constructor() {
+    this._root = null
+  }
+
+  contains(data) {
+    let foundNode = false
+    let currentNode = this._root
+
+    while(!foundNode && currentNode) {
+      if (data < currentNode.data) {
+        currentNode = currentNode.left
+      } else if (data > currentNode.data) {
+          currentNode = currentNode.right
+      } else {
+          foundNode = true;
+      }
+    }
+    return foundNode
+  }
+
+  insert(data) {
```

Comment:
could you possibly compartmentalize some of this logic in this function? Possibly refactor to move some of this logic out of the same function? 

https://en.wikipedia.org/wiki/Single_responsibility_principle 👈